src/ephemeris/set_library_permissions.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+'''Tool to set permissions for all datasets of a given Galaxy Data Library'''
+
+import argparse
+import logging as log
+from bioblend import galaxy
+import sys, time, os
+from rich.progress import Progress
+from .common_parser import get_common_args
+# Print iterations progress
+
+
+def get_datasets(gi, library_id) -> [str]:
+    objects = gi.libraries.show_dataset(library_id=library_id, dataset_id='')
+    datasets = []
+    for index in range(len(objects)):
+       if objects[index]['type'] == 'file':
+           datasets.append(objects[index]['id'])
+    if datasets == []:
+        sys.exit("No datasets in library!")
+    else:
+        return datasets
+
+def set_permissions(gi, library_id, role_ids):
+    log.info("Your library_id is " + library_id + "\n")
+    log.info("Your roles are: %s", " ".join(role_ids))
+    datasets = get_datasets(gi, library_id)
+    total = len(datasets)
+    est = total*3/60
+    # Give User time to abort
+    log.info('\nSuccess! %d datasets found. Processing can take up to %f min', total, est)
+    if input("Do you want to continue? (y/n) ") == "y":
```

Comment:
A lot of ephemeris is meant to be used as a library, it's unfortunate to have `input` in that case. Can you make this optional ?

Reply:
I could add a -y flag, would that be okay?

Reply:
if the function's default argument is to be quiet, yes.